tests/bearlib/languages/documentation/DocumentationCommentTest.py

```diff
@@ -34,3 +37,101 @@ def test_fields(self):
+                    self.Param(name='test', desc=' test description2 \n')]
+        self.check_docstring(doc, expected)
+
+    def test_return_values(self):
+        doc = (" :return: something1 \n"
+               " @return: something2 ")
+        expected = [self.RetVal(desc=' something1 \n'),
+                    self.RetVal(desc=' something2 ')]
+        self.check_docstring(doc, expected)
+
+    def test_docstring_file(self):
+        data = self.load_testdata("data2.py")
+
+        language = "python"
+        docstyle = "doxygen"
+
+        docs = tuple(extract_documentation(data, language, docstyle))
+
+        parsed_docs = list()
+        markers = list()
+
+        for doc in docs:
+            markers.append(doc.marker)
+            parsed_docs.append(doc.parse_documentation())
+
+        expected = [
+            [self.Desc(desc='\nModule description.\n\n'
+                            'Some more foobar-like text.\n')],
+            [self.Desc(desc='\nA nice and neat way of documenting code.\n'),
+             self.Param(name='radius', desc='The explosion radius.\n')],
+            [self.Desc(desc='\nA function that returns 55.\n')],
+            [self.Desc(desc='\nDocstring with layouted text.\n\n    layouts'
+                            ' inside docs are preserved for these'
+                            ' documentation styles.\nthis is intended.\n')],
+            [self.Desc(desc=' Docstring directly besides triple quotes.\n'
+                            '    Continues here. ')],
+            [self.Desc(desc=' Alternate documentation style in doxygen.\n '
+                            ' Subtext\n More subtext (not correctly aligned)\n'
+                            '      sub-sub-text\n\n')],
+            [self.Desc(desc='\nThis is the best docstring ever!\n'),
+             self.Param(name='param1:', desc='    Very Very Long Parameter '
+                                             'description.\n'),
+             self.Param(name='param2:', desc='    Short Param description.\n'),
+             self.RetVal(desc='Long Return Description That Makes No Sense'
+                              ' And Will\n         Cut to the Next Line.\n')]]
+
+        self.assertEqual(parsed_docs, expected)
+
+    def check_docstring(self,
+                        docstring,
+                        expected=[]):
+        self.assertIsInstance(docstring,
+                              str,
+                              "docstring needs to be a string for this test.")
+
+        self.assertIsInstance(expected,
+                              list,
+                              "docstring needs to be a list for this test.")
+
+        doc_comment = DocumentationComment(docstring, "python", "default")
+        parsed_metadata = doc_comment.parse_documentation()
+        self.assertEqual(parsed_metadata, expected)
+
+    def test_not_implemented(self):
+        # TODO: make this more practical
```

Comment:
I think you should rather use `FIXME` then, that identifies something more "long-term". But problem is as soon as we implement stuff for VHDL stuff breaks, this is not cool and would confuse the one that implements VHDL markers^^